tarantool_test.go

```diff
@@ -92,6 +92,37 @@ func BenchmarkClientSerial(b *testing.B) {
 	}
 }
 
+func BenchmarkClientSerialRequestObject(b *testing.B) {
```

Comment:
Probably it would be helpful to add a Benchmark with the parallel load, where you call b.RunParallel inside. That would show the full difference between the existing API and the proposed one. Feel free to ignore, I have already implemented it in scope of my patch, which is over your one. On the other hand my patch could be discarded and we will lose those Benchmarks.

Reply:
I don't think that request objects will somehow critically affect parallel execution in comparison with single-threaded. The benchmark will look better in the issue with contexts.